Library/Contributions/pre-push

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/python
+# Makes sure bottles are fetchable after BrewTestBot modifies a formula.
+# Only checks pushes to master on Homebrew/homebrew* remotes.
+# To use, `ln -s $(brew --repository)/Library/Contributions/pre-push
+# $(brew --repository)/.git/hooks`.
+
+import re
+import os
+import subprocess
+import sys
+
+FORMULA_RE = re.compile(r"^(Library/Formula/)?([A-Za-z0-9-]+)\.rb")
+TESTBOT_EMAILS = ["brew-test-bot@googlegroups.com"]
+
+def git(*args):
+    return subprocess.check_output(["git"] + list(args)).strip()
+
+def commit_author_email(commit_sha):
+    return git("show", "-s", "--pretty=%ae", commit_sha)
+
+def files_touched_by_commit(commit_sha):
+    return git("diff-tree", "--name-only", "-r", "--no-commit-id", commit_sha).split("\n")
+
+def commits_between(beginning, end):
+    return git("rev-list", "{}..{}".format(beginning, end)).split("\n")
+
+def push_okay(remote_name, remote_url, updates_file):
+    if "omebrew/homebrew" not in remote_url:
+        return True
+    updates = [line.strip().split() for line in updates_file.readlines()]
```

Comment:
True, but the input will be a single line in the typical case and I like the list unpacking idiom (`for x, y, z in list_of_lists:`) though I'm willing to hear that I shouldn't